### ISSUE TITLE
Is it possible to use Stonefish for surface vehicles/boats?

### ISSUE THREAD

**Issue by therishidesai** [patrykcieslak/stonefish#32]
I want to be able to use the stonefish simulator for boats and I was wondering if that has ever been tried.

**Comment by silent-code** [comment]
Yes, there is a minimal example in the tests folder: https://github.com/patrykcieslak/stonefish/blob/master/Tests/FloatingTest/FloatingTestManager.cpp

**Comment by patrykcieslak** [comment]
Yes it is possible, the buoyancy is working realistically, so if you build a boat it will not sink. However, I have not worked on surface vehicles yet but I know there is one strange behavior that makes all things that are on the surface rotate slowly. I will try to fix it as I am implementing some stuff now.

**Comment by Hedjet** [comment]
Hi, I'm working right now on a large surface vehicle in Stonefish and it works fine except for the mentioned behavior that makes the vehicle have a constant angular rotation in Z... I tried many options in the environment and vehicle physics with no success. Is there any possible solution or a future fix?
